src/marshmallow/fields.py

```diff
@@ -1526,7 +1528,7 @@ def _serialize(self, value, attr, obj, **kwargs):
 
     def _deserialize(self, value, attr, data, **kwargs):
         try:
-            value = self.serialization_type(value)
+            value = float(value)
```

Comment:
Is it possible that when deserializing an `int` this introduces an error due to floating point precision?

Like `float(42)` -> `42.0000000000000000001`

Answering to myself, no because this will be rounded in timedelta:

```
>>> field.deserialize(12.9000000001)
datetime.timedelta(seconds=12, microseconds=900000)
```